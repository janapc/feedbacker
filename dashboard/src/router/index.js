import { createRouter, createWebHistory } from 'vue-router';

const Home = () => import('../views/Home/index.vue');
const Feedbacks = () => import('../views/Feedbacks/index.vue');
const Credencials = () => import('../views/Credencials/index.vue');

export const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home,
  },
  {
    path: '/feedbacks',
    name: 'Feedbacks',
    component: Feedbacks,
    meta: {
      hasAuth: true,
    },
  },
  {
    path: '/credencials',
    name: 'Credencials',
    component: Credencials,
    meta: {
      hasAuth: true,
    },
  },
  {
    path: '/:pathMatch(.*)*',
    redirect: { name: 'Home' },
  },
];

const router = createRouter({
  history: createWebHistory('/'),
  routes,
});

export default router;
