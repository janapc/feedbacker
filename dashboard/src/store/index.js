import { readonly } from 'vue';
import UserModule from './user';
import GlobalModule from './global';

export default readonly({
  User: UserModule,
  Global: GlobalModule,
});
