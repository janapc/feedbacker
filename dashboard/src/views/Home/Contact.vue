<template>
  <div class="flex justify-center w-full">
    <div class="flex flex-col items-center w-4/5 max-w-6xl my-16">
      <h1 class="text-4xl font-black text-center text-gray-800">Alguma dúvida?</h1>
      <p class="text-lg text-center text-gray-800 font-regular">
        Quer saber melhor como funciona e quais são os preços?
      </p>
      <div class="mt-10">
        <a
          href="mailto:"
          class="px-6 py-2 mt-10 font-bold text-white rounded-full bg-brand-main focus:outline-none"
          >Nos mande um e-mail</a
        >
      </div>
    </div>
  </div>
</template>

<script>
export default {};
</script>
