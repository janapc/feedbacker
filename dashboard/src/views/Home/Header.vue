<template>
  <header class="header">
    <div class="header-group">
      <div class="flex items-center justify-between py-10">
        <div class="w-28 lg:w-36">
          <img src="../../assets/images/logo_white.png" alt="logo" class="w-full" />
        </div>

        <div class="flex">
          <button
            id="header-create-account-button"
            @click="() => emit('create-account')"
            class="px-6 py-2 font-bold rounded-full text-white focus:outline-none"
          >
            Crie uma conta
          </button>

          <button
            id="header-login-button"
            @click="() => emit('login')"
            class="px-6 py-2 font-bold bg-white rounded-full text-brand-main focus:outline-none"
          >
            Entrar
          </button>
        </div>
      </div>

      <div class="flex flex-col mt-28">
        <h1 class="text-4xl font-black text-white">
          Tenha um feedback.<br />
          E faça seus clientes mais <br class="hidden lg:inline-block" />
          felizes!
        </h1>

        <p class="text-lg font-medium text-white">
          Receba ideias, reclamações e feedbacks com um <br class="hidden lg:inline-block" />
          simples widget na página.
        </p>

        <div>
          <button
            @click="() => emit('create-account')"
            id="cta-create-account-button"
            class="px-6 py-2 mt-10 font-bold bg-white rounded-full text-brand-main focus:outline-none"
          >
            Crie uma conta grátis
          </button>
        </div>
      </div>
    </div>
  </header>
</template>

<script>
export default {
  setup(props, { emit }) {
    return { emit };
  },
};
</script>

<style lang="postcss" scoped>
.header {
  @apply bg-brand-main w-full flex justify-center;
  height: 700px;
}

.header-group {
  @apply flex flex-col w-4/5 max-w-6xl;
}
@media (min-width: 640px) {
  .header-group {
    background-image: url("../../assets/images/blue-balloons.png");
    background-size: 628px;
    background-position: 90% 100%;
    background-repeat: no-repeat;
  }
}
</style>
