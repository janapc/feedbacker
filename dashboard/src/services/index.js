import axios from 'axios';

import router from '../router';
import AuthService from './auth';
import UsersService from './users';
import FeedbacksService from './feedbacks';

import { setGlobalLoading } from '../store/global';

const API_ENVS = {
  production: 'https://backend-vue3js-ixvpypm03.vercel.app/',
  development: 'http://localhost:3000',
  local: 'http://localhost:3000',
};

const httpClient = axios.create({
  baseURL: API_ENVS[process.env.NODE_ENV] ?? API_ENVS.local,
});

httpClient.interceptors.response.use(
  (response) => {
    setGlobalLoading(false);
    return response;
  },
  (error) => {
    const canThrowAnError = error.request.status === 0 || error.request.status === 500;

    if (canThrowAnError) {
      setGlobalLoading(false);
      throw new Error(error.message);
    }
    if (error.response.status === 401) {
      router.push({ name: 'Home' });
    }
    setGlobalLoading(false);
    return error;
  },
);

httpClient.interceptors.request.use((config) => {
  setGlobalLoading(true);
  const token = window.localStorage.getItem('token');

  if (token) {
    config.headers.common.Authorization = `Bearer ${token}`;
  }
  return config;
});

export default {
  auth: AuthService(httpClient),
  users: UsersService(httpClient),
  feedbacks: FeedbacksService(httpClient),
};
