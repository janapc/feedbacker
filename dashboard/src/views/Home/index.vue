<template>
  <Header id="header-component" @create-account="handleAccountCreate" @login="handleLogin" />
  <Contact />

  <div class="flex justify-center py-10 bg-brand-gray">
    <p class="font-medium text-center text-gray-800">feedbacker 2021</p>
  </div>
</template>

<script>
import { onMounted } from 'vue';
import { useRouter } from 'vue-router';

import Header from './Header.vue';
import Contact from './Contact.vue';

import useModal from '../../hooks/useModal';

export default {
  components: {
    Header,
    Contact
  },
  setup () {
    const router = useRouter();
    const modal = useModal();

    onMounted(() => {
      const token = window.localStorage.getItem('token');

      if (token) router.push({ name: 'Feedbacks' });
    });

    function handleLogin () {
      modal.open({
        component: 'ModalLogin'
      });
    }

    function handleAccountCreate () {
      modal.open({
        component: 'ModalCreateAccount'
      });
    }

    return {
      handleLogin,
      handleAccountCreate
    };
  }
};
</script>

<style></style>
