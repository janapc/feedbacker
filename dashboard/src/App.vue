<template>
  <modal-factory />
  <router-view />
</template>

<script>
import { watch } from 'vue';
import { useRoute, useRouter } from 'vue-router';

import ModalFactory from './components/ModalFactory/index.vue';
import services from './services';
import { setCurrentUser } from './store/user';

export default {
  components: { ModalFactory },
  setup() {
    const route = useRoute();
    const router = useRouter();

    watch(
      () => route.path,
      async () => {
        if (route.meta.hasAuth) {
          const token = window.localStorage.getItem('token');

          if (!token) {
            router.push({ name: 'Home' });
          }

          const { data } = await services.users.getMe();

          setCurrentUser(data);
        }
      },
    );
  },
};
</script>
