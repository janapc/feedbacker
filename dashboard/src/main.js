import { createApp } from 'vue';
import Toast, { POSITION } from 'vue-toastification';
import App from './App.vue';
import router from './router';

import 'animate.css';

import '@/assets/css/tailwind.css';
import '@/assets/css/fonts.css';
import 'vue-toastification/dist/index.css';

const app = createApp(App);
app.use(router);
app.use(Toast, { position: POSITION.BOTTOM_RIGHT });
app.mount('#app');
