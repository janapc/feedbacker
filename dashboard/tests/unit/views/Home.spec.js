import { createRouter, createWebHistory } from 'vue-router';
import { shallowMount, flushPromises } from '@vue/test-utils';

import Home from '../../../src/views/Home/index.vue';
import { routes } from '../../../src/router';

const router = createRouter({
  history: createWebHistory('/'),
  routes
});

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn()
};
global.localStorage = localStorageMock;

describe('<Home />', () => {
  it('should render home correctly', async () => {
    router.push('/');
    await router.isReady();
    const wrapper = shallowMount(Home, {
      global: {
        plugins: [router]
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should go to page feedback if the user has a token', async () => {
    localStorage.setItem('token', '123');
    router.push('/');

    await router.isReady();
    shallowMount(Home, {
      global: {
        plugins: [router]
      }
    });

    expect(localStorage.getItem('token')).toBe('123');
    await flushPromises();

    expect(router.currentRoute.value.name).toBe('Feedbacks');

    localStorage.clear();
  });
});
