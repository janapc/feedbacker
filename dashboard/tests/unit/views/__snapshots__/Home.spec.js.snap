// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render home correctly 1`] = `
<header-stub id="header-component"></header-stub>
<contact-stub></contact-stub>
<div class="flex justify-center py-10 bg-brand-gray">
  <p class="font-medium text-center text-gray-800">feedbacker 2021</p>
</div>
`;
